io/disk/mvcli_test.py

```diff
@@ -0,0 +1,267 @@
+        :param pd_ids: List of phyisical disk ID's under adapter id adapter_id
+        """
+        if 'ppc64le' not in platform.processor():
+            self.cancel("Processor is not ppc64le")
+        self.tool_url = self.params.get('tool_url', '*', default=None)
+        self.adapter_id = self.params.get('adapter_id', '*', default=None)
+        self.pd_ids = self.params.get('pd_ids', '*', default=None).split(",")
+        path = self.fetch_asset("mvcli", locations=[self.tool_url],
+                                expire="7d")
+        self.run_command("cp %s %s" % (path, self.teststmpdir))
+        mvcli_path = "%s/mvcli" % self.teststmpdir
+        self.run_command("chmod +x %s" % mvcli_path)
+        self.base = "%s %s" % (mvcli_path, self.adapter_id)
+        self.base_force = "echo y | %s %s" % (mvcli_path, self.adapter_id)
+        self.run_command("dmesg -c")
+
+    def run_command(self, cmd):
+        """
+        Run command and fail the test if any command fails
+        """
+        try:
+            process.run(cmd, shell=True, sudo=True)
+        except CmdError as details:
+            self.fail("Command %s failed %s" % (cmd, details))
+
+    def run_cmd_output(self, cmd):
+        """
+        Execute the command and return output
+        """
+        output = process.system_output(cmd, ignore_status=True,
+                                       shell=True, sudo=True)
+        return output
+
+    def test_help(self):
+        """
+        Get brief help for all commands or detail help for one command.
+        """
+        self.log.info("Mvcli help test")
+        self.run_command("%s help" % self.base)
+
+    def test_info(self):
+        """
+        Display adapter(hba), virtual disk(vd), disk array,
+               physical disk(pd), Port multiplexer(pm), expander(exp),
+               block disk(blk) or spare drive information.
+        """
+        self.log.info("Mvcli info test")
+        objects = ["hba", "pd", "vd", "pm", "blk"]
+        for obj in objects:
+            self.run_command("%s info -o %s" % (self.base, obj))
+        for pd in self.pd_ids:
+            self.run_command("%s info -o pd -i %s" % (self.base, pd))
+
+    def test_get(self):
+        """
+        Get configuration information of VD, PD, Array, HBA or Driver.
+        """
+        self.log.info("Mvcli get test")
+        self.run_command("%s get -o hba" % self.base)
+        for pd in self.pd_ids:
+            self.run_command("%s get -o pd -i %s" % (self.base, pd))
+
+    def test_set(self):
+        """
+        Set configuration parameters of VD, PD, HBA or Driver.
+        """
+        self.log.info("Mvcli set configuration parameters test")
+        self.run_command("%s set -o hba -i %s --rawupdateon" % (self.base,
+                         self.adapter_id))
+        get_cmd = "%s get -o hba -i %s" % (self.base, self.adapter_id)
+        output = self.run_cmd_output(get_cmd)
+        if "Raw Update:              on" in output:
```

Comment:
This is not the best parsing. My suggestion:

`if "Raw Update:" in output and "on" in output:`